### PR TITLE
[SPARK-31181][SQL][TESTS] Remove the default value assumption on CREATE TABLE test cases

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

A few `CREATE TABLE` test cases have some assumption on the default value of `LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED`. This PR (SPARK-31181) makes the test cases more explicit from test-case side.

The configuration change was tested via https://github.com/apache/spark/pull/27894 during discussing SPARK-31136. This PR has only the test case part from that PR.

### Why are the changes needed?

This makes our test case more robust in terms of the default value of `LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED`. Even in the case where we switch the conf value, that will be one-liner with no test case changes.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass the Jenkins with the existing tests.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala [hunk 1]
```diff
@@ -40,7 +40,9 @@ import org.apache.spark.sql.test.SharedSparkSession
 import org.apache.spark.sql.types.{IntegerType, StructField, StructType}
 
 class DDLParserSuite extends AnalysisTest with SharedSparkSession {
-  private lazy val parser = new SparkSqlParser(new SQLConf)
+  private lazy val parserConf = new SQLConf
+  parserConf.setConf(SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED, false)
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala]: This test suite heavily depends on `LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED=false`.

**Comment by dongjoon-hyun** [comment]
Hi, @cloud-fan . Could you review this PR?
At least, I believe we need this PR for the downstream who consumes our repository.

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/create_view.sql [hunk 2]
```diff
@@ -41,7 +41,7 @@ DROP TABLE emp;
 -- These views are left around mainly to exercise special cases in pg_dump.
 
 -- [SPARK-19842] Informational Referential Integrity Constraints Support in Spark
-CREATE TABLE view_base_table (key int /* PRIMARY KEY */, data varchar(20));
+CREATE TABLE view_base_table (key int /* PRIMARY KEY */, data varchar(20)) USING PARQUET;
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/postgreSQL/create_view.sql]: Shall we keep the pgsql test as what it is in pgsql?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala]: nit:
```
private lazy val parser = new SparkSqlParser(new SQLConf().copy(
  SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED -> false))
```

File: sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala [hunk 3]
```diff
@@ -844,18 +844,20 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
...
-      sql("create table tab2 ( a int)")
-      sql("insert into tab2 values(2)")
-      checkAnswer(sql("select * from tab2"), Seq(2).map(i => Row(i)))
-      sql(s"""insert overwrite local directory '$path' using parquet select * from tab2""")
-      sql("refresh table tab1")
-      checkAnswer(sql("select * from tab1"), Seq(2).map(i => Row(i)))
+    withSQLConf(SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "false") {
+      withTempPath { dir =>
+        val path = dir.toURI.getPath
+        sql(s"""create table tab1 ( a int) location '$path'""")
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala]: This is to test data source table, maybe just add USING?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala [hunk 4]
```diff
@@ -1520,13 +1520,15 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
...
-             | CREATE TABLE $ext_tbl (key INT, value STRING, ds STRING)
-             | PARTITIONED BY (ds)
-             | LOCATION '${dir.toURI}'
-           """.stripMargin)
+        withSQLConf(SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "false") {
+          sql(s"CREATE TABLE $tbl (key INT, value STRING, ds STRING) PARTITIONED BY (ds)")
+          sql(
+            s"""
+               | CREATE TABLE $ext_tbl (key INT, value STRING, ds STRING)
+               | PARTITIONED BY (ds)
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala]: `PARTITIONED BY (ds)` implicitly indicates data source table, maybe just add USING?

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/create_view.sql (hunk 2 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/create_view.sql]: Hi, @cloud-fan . We are using `CREATE TABLE USING` even in `pgsql` test. It's okay to change this.

File: sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala (hunk 3 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala]: Sure~

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala (hunk 4 above)
Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala]: Sure~

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala]: Thanks!

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan ! Also, I closed #27894.

**Pull request closed without merging**